src/components/editor/Editor.js

```diff
@@ -29,6 +29,24 @@ class Editor extends Component {
         } catch (error) {
             console.error("Unable to attach custom completers");
         }
+
+        // Warn the issue before refreshing the page
+        window.addEventListener("beforeunload", (event) => {
```

Comment:
Does this prevent the user from closing the editor with unsaved changes (but not leave the page)?

Reply:
beforeunload will only fire when the user is leaving the page with unsaved change.